scripts/process_data_dcm-zurich.sh

```diff
@@ -205,6 +249,49 @@ else
         # solidity
         sct_compute_compression -i ${file_t2_ax_seg}.nii.gz -vertfile ${file_t2_ax_seg}_labeled.nii.gz -l ${file_compression}.nii.gz -normalize-hc 1 -sex ${sex} -metric solidity -o ${PATH_RESULTS}/compression_metrics.csv
 
+        # ------------------------------------------------------------------------------
+        # Adding new metrics 
+        # ------------------------------------------------------------------------------
+
+        # ------------------------------------------------------------------------------
+        # Compute general morphometric metrics across all vertebral levels
+        # ------------------------------------------------------------------------------
+        echo "Computing general morphometric metrics across vertebral levels..."
+        
+        # Check if vertebral-level metrics CSV already exists to avoid reprocessing
+        # WOUld this name be the standard name for vertebral-level metrics CSV?
+        vertebral_metrics_csv="${PATH_RESULTS}/vertebral_level_metrics.csv"
+
+        # Compute CSA, AP, RL, eccentricity and solidity across vertebral levels if they are not already available
+        sct_process_segmentation -i ${file_t2_ax_seg}.nii.gz  -discfile ${file_t2_ax_labels}.nii.gz -perlevel 1 -o ${PATH_RESULTS}/vertebral_level_metrics_cord.csv -append 1
+        # Normalized to PAM50 
+        sct_process_segmentation -i ${file_t2_ax_seg}.nii.gz  -discfile ${file_t2_ax_labels}.nii.gz -normalize-PAM50 1 -perslice 1 -perlevel 1 -o ${PATH_RESULTS}/vertebral_level_metrics_cord_normalized.csv -append 1
+
+        # Always try to segment canal (check for manual first, then automatic)
+        segment_canal_if_does_not_exist ${file_t2_ax} 't2'
+        file_t2_ax_canal_seg=$FILESEG
+
+        # Compute CSA, AP, RL for canal segmentation 
+        if [[ -e ${file_t2_ax}_label-canal_mask.nii.gz ]]; then
+          echo "Computing CSA, AP, RL across vertebral levels for canal segmentation..."
+          # Compute CSA, AP, RL across vertebral levels for canal segmentation
+          vertebral_metrics_canal_csv="${PATH_RESULTS}/vertebral_level_metrics_canal.csv"
+          sct_process_segmentation -i ${file_t2_ax_canal_seg}.nii.gz -discfile ${file_t2_ax_labels}.nii.gz -perlevel 1 -o ${vertebral_metrics_canal_csv} -append 1
+        fi
+
+        # Always try to segment lesion
+        segment_lesion_if_does_not_exist ${file_t2_ax} 't2'
+        file_t2_ax_lesion_seg=$FILESEG
+        #Compute statistics on segmented lesions if lesion segmentation was performed
+        if [[ -e ${file_t2_ax}_label-lesion_mask.nii.gz ]]; then
```

Comment:
probably not necessary as your function `segment_lesion_if_does_not_exist ` already does this check